packages/core/src/generate-anchor.ts

```diff
@@ -75,68 +42,138 @@ export function displayMarkers(anchors, activeStep: number, showAllHints: boolea
           y: a.coords.y + textOffset
         });
       }
-
-    // Find the positioning only if we provided no coords
-    } else {
+    } else { // Find the positioning only if we provided no coords
       const svgPosition = visElement.getBoundingClientRect();
 
       let node;
       if(isOnboardingElementAnchor(a)) {
         node = a.element;
       } else {
-        const elToAppendTo = d3.select(a.sel);
-        if(!elToAppendTo.node()) {
+        const elToAppendTo = document.querySelector(a.sel);
+        if(!elToAppendTo) {
           console.error('No element found for selector', a.sel);
           return;
         }
-        node = elToAppendTo.node()
+        node = elToAppendTo;
       }
       const elRect = node.getBoundingClientRect();
       const elBox = node.getBBox();
-      // console.log('FOR ', a.sel || a.element ,' the DOMRect = ', elRect, ' and the SVGrect = ', elBox);
       Object.assign(settings, {
-        cx: elRect.x - svgPosition.left,
-        cy: elRect.y - svgPosition.top,
+        cx: elRect.x,
+        cy: elRect.y,
         r,
-        x: elRect.x - svgPosition.left,
-        y: (elRect.y - svgPosition.top) + textOffset,
+        x: elRect.x,
+        y: (elRect.y) + textOffset,
       });
     }
+    Object.assign(settings, { 
+      color: getColor(stage),
+      clickEvent: clickEvent
+    })
     // Create the respective anchor
-    createHint(settings, i, activeStep, showAllHints);
+    createHint(settings, text, message, title, index); 
+    text++;
+    if (stage !== anchors[index + 1]?.stage) text = 1;
   });
 }
 
 /**
- * Somewhat generic function to create an annotation based on some properties that can vary.
- * @param {*} settings where all the positions for the annotation are passed
- * @param {*} text of the annotation to show
+ * Somewhat generic function to create the hints based on some properties that can vary.
+ * @param {*} settings where all the positions, colors and clickevents for the anchor are passed
+ * @param {*} text of the anchor to show
+ * @param {*} message tooltip message for anchor
+ * @param {*} title title of tooltip
+ * @param {*} index index of anchor
  */
-function createHint(settings, text, activeStep: number, showAllHints: boolean) {
-  let { cx, cy, r, x, y, left, right, top, bottom } = settings;
-  const hG = d3
-    .select('.onboardingAnnotations')
-    // .html(null)
-    .append('g')
-    .classed('chartAnnotation', true)
-    .classed('active', showAllHints || activeStep === text-1 ? true : false)
-    .attr('id', `anchor-${text}`);
+function createHint(settings, text, message, title, index) { //unused params: activeStep: number, showAllHints: boolean
```

Comment:
```suggestion
function createHint(settings: ??, anchorText: string, anchorIndex: number tooltipTitle: string, tooltipMessage: string) {
```

Reorder and type parameters.